### PR TITLE
Remove the ember-simple-auth mixins

### PR DESCRIPTION
**Pull request by Windvis**
This replaces the ember-simple-auth mixins with session service based alternatives.

### CODE REVIEW

File: app/routes/login.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
-export default Route.extend({
-});
+export default class LoginRoute extends Route {
+  @service session;
+
+  beforeModel() {
+    this.session.prohibitAuthentication('index');
```
Comment (Windvis) [inline app/routes/login.js:8]: The previous version didn't redirect when the user was already logged in.

File: app/routes/import.js [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
-  authenticationRoute: computed('mock', function() {
-    if (this.mock) {
-      return 'mock-login';
-    }
-    else {
-      return 'login'; // TODO read from configuration
-    }
-  })
-});
+    let loginRoute = transition.to.queryParams.mock ? 'mock-login' : 'login';
```
Comment (Windvis) [inline app/routes/import.js:13]: This is the only route that has this kind of logic, but I'm not sure if it is still needed?

Comment (nvdk) [reply app/routes/import.js:13]: ah it does! this is actually a route other applications can redirect to if they want to pass information to GN and this way we can also transition to the correct route after login in testing scenario's.

Comment (Windvis) [reply app/routes/import.js:13]: Aha ok! I'll leave it like this then.

**Review by nvdk — APPROVED** [review]
Looks to be working fine! I guess we can now also (re)enable the mixin linting rule :+1:

**Comment by Windvis** [comment]
> Looks to be working fine! I guess we can now also (re)enable the mixin linting rule +1

@nvdk There are still some mu-semtech/ember-data-table mixins in use. Or do you mean enable it and add disable comments to those specific files?

**Comment by nvdk** [comment]
right I keep forgetting about ember-data-table, we really should get around to modernizing that addon :). I guess we keep the rule as is for now, I don't see much benefit in disabling in the specific files as it's rather unlikely we'll introduce new mixins anyway.

**Pull request merged**